library/ecp.c

```diff
@@ -382,7 +390,12 @@ int mbedtls_ecp_copy( mbedtls_ecp_point *P, const mbedtls_ecp_point *Q )
  */
 int mbedtls_ecp_group_copy( mbedtls_ecp_group *dst, const mbedtls_ecp_group *src )
 {
-    return mbedtls_ecp_group_load( dst, src->id );
+    /* Validate input parameters*/
+    if ( dst == NULL || src == NULL )
+        return ( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
+    if ( dst == src )
+        return ( 0 );
+    return mbedtls_ecp_group_load(dst, src->id);
```

Comment:
Spacing as above, and (pre-existing defect) there should be a bracket around the call to `mbedtls_ecp_group_load`.